request_session/request_session.py

```diff
@@ -404,7 +401,16 @@ def _process(
                             if response_text != ""
                             else None
                         )
-                        self.sentry_client.captureException(extra=extra_data)
+                        if hasattr(
+                            self.sentry_client, "capture_exception"
+                        ) and callable(
+                            getattr(self.sentry_client, "capture_exception")
+                        ):
+                            self.sentry_client.capture_exception(extras=extra_data)
+                        elif hasattr(
+                            self.sentry_client, "captureException"
+                        ) and callable(getattr(self.sentry_client, "captureException")):
```

Comment:
🤔 can it happen that `capture_exception` or `captureException` aren't callables? either way... what do you think about using the third argument that `getattr` accepts to pass a non-callable default and only check for callable?

```suggestion
                        if callable(getattr(self.sentry_client, "capture_exception", None)):
                            self.sentry_client.capture_exception(extras=extra_data)
                        elif callable(getattr(self.sentry_client, "captureException", None)):
```

Reply:
also... it shouldn't happen but what if both conditions are false? 😅️
wdyt about raising an exception? something to let the user know that he's lacking a sentry client

Reply:
I'll update code with your three argument getattr usage. But I am not sure about raising in error in `_process` method, it can happen only if sentry is passed misconfigured and as such, I think a better place would be to pass raise an error on initialization. That would blow up on mypy anyway, but I guess we should not expect clients to run mypy checks.

I would keep it as it is now and if we think we should do better validation of passed args/objects on init, I would handle it in different PR.

Reply:
>That would blow up on mypy anyway
Not everyone uses mypy tho 😅️

but agree, let's keep this out of scope for now 🙂️
thanks!